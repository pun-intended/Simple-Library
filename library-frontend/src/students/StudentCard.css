.StudentCard {
    min-height: 6vh;
    margin-top: 1vh;
    margin-bottom: 1vh;
}