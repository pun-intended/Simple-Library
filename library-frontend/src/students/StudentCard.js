// Name, class, Book cover if they have checked out
// book - onClick ->BookDetails
import React, {useState} from "react"
import { Row, Col, Card, CardBody, CardTitle, CardText, Button } from 'reactstrap'
import CheckOutStudent from "./CheckOutStudent"
import CheckInModal from "../books/CheckInModal"


const StudentCard = ({student, setUpdate}) => {
    
    const [outModal, setOutModal] = useState(false);
    const [inModal, setInModal] = useState(false);
    const toggleOut = () => { setOutModal(!outModal)};
    const toggleIn = () => { setInModal(!inModal)};
    
    return(
        <Card className="StudentCard">
            <Row>
            <Col >
            <CardTitle tag="h6">{student.first_name} {student.last_name}</CardTitle>
            </Col>
            <Col>
            {student.book_id && 
                <Button className="btn" color="secondary" onClick={toggleIn}> Check in </Button>
            }

            {!student.book_id &&
                <Button className="btn" color="primary" onClick={toggleOut}> Check out </Button>}
            {outModal && <CheckOutStudent modal={outModal} toggle={toggleOut} student={student} setUpdate={setUpdate}/>}
            {inModal && <CheckInModal modal={inModal} toggle={toggleIn} book_id={student.book_id} setUpdate={setUpdate}/>}
            </Col>
            </Row>
        </Card>
        
    )
}

export default StudentCard;