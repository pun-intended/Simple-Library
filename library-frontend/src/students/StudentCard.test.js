import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import StudentCard from "./StudentCard";


const setUpdate = jest.fn()
const toggleOut = jest.fn()
// const toggleIn = jest.fn()
const studentHas = {
    first_name: "test",
    last_name: "student",
    book_id: "101",
};
const studentHasNot = {
    first_name: "test",
    last_name: "student",
};
// Smoke test
it("renders without crashing", () => {
    render(<StudentCard student={studentHas} setUpdate={setUpdate} />)
})

// Snapshot
it("matches snapshot", () => {
    const {asFragment} = render(<StudentCard student={studentHas} setUpdate={setUpdate} />);
    expect(asFragment()).toMatchSnapshot();
})


// Check in button opens modal
it("opens check in modal when check in clicked", () => {
    render(<StudentCard student={studentHas} setUpdate={setUpdate} />)
    const btn = screen.getAllByText(/check in/i)
    fireEvent.click(btn[0])
    expect(screen.getByRole('dialog')).toBeInTheDocument()
})

// // check_out button calls toggleOut method
it("opens check in modal when check in clicked", () => {
    render(<StudentCard student={studentHasNot} setUpdate={setUpdate} />)
    const btn = screen.getAllByText(/check out/i)
    fireEvent.click(btn[0])
    expect(screen.getByRole('dialog')).toBeInTheDocument()
});

// // card contains student name
it("Contains student name", () => {
    render(<StudentCard student={studentHas} setUpdate={setUpdate} />)
    expect(screen.getByText(/test student/i)).toBeInTheDocument()
})

// Shows check out button for student without book
it("Shows check out button for student without book", () => {
    render(<StudentCard student={studentHasNot} setUpdate={setUpdate} />)
    expect(screen.getByText(/check out/i)).toBeInTheDocument()
})


// Shows check in button for student with book
it("Shows check in button for student with book", () => {
    render(<StudentCard student={studentHas} setUpdate={setUpdate} />)
    expect(screen.getByText(/check in/i)).toBeInTheDocument()
})