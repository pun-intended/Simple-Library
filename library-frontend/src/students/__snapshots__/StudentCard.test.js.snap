// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="StudentCard card"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h6
          class="card-title"
        >
          test student
        </h6>
      </div>
      <div
        class="col"
      >
<<<<<<< Updated upstream
        <svg
          aria-labelledby="svg-inline--fa-title-CUpsAwnhoJs2"
          class="svg-inline--fa fa-book-open fa-lg btn btn-primary"
          data-icon="book-open"
          data-prefix="fas"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="svg-inline--fa-title-CUpsAwnhoJs2"
          >
            has-book
          </title>
          <path
            d="M249.6 471.5c10.8 3.8 22.4-4.1 22.4-15.5V78.6c0-4.2-1.6-8.4-5-11C247.4 52 202.4 32 144 32C93.5 32 46.3 45.3 18.1 56.1C6.8 60.5 0 71.7 0 83.8V454.1c0 11.9 12.8 20.2 24.1 16.5C55.6 460.1 105.5 448 144 448c33.9 0 79 14 105.6 23.5zm76.8 0C353 462 398.1 448 432 448c38.5 0 88.4 12.1 119.9 22.6c11.3 3.8 24.1-4.6 24.1-16.5V83.8c0-12.1-6.8-23.3-18.1-27.6C529.7 45.3 482.5 32 432 32c-58.4 0-103.4 20-123 35.6c-3.3 2.6-5 6.8-5 11V456c0 11.4 11.7 19.3 22.4 15.5z"
            fill="currentColor"
          />
        </svg>
=======
        <button
          class="btn btn-primary btn btn-secondary"
          type="button"
        >
           Check in 
        </button>
>>>>>>> Stashed changes
      </div>
    </div>
  </div>
</DocumentFragment>
`;
